### PR TITLE
Handle squid rout errors

### PR DESCRIPTION
**Pull request by rosepuppy**
Screenshot 2023-11-08 at 1 42 05 PM

- Handle squid errors passed back via abacus
- `src/lib/abacus/rest.ts`
  - Do not filter response of requests with unsuccessful status codes
- fix max deposit button
- disable slippage editor for now, as we can't edit slippage in abacus yet

### CODE REVIEW

File: src/lib/abacus/rest.ts [hunk 1]
```diff
@@ -56,13 +56,8 @@ class AbacusRest implements AbacusRestProtocol {
       .then(async (response) => {
         const data = await response.text();
 
-        if (response.ok) {
-          callback(data, response.status);
-        } else {
-          // response not OK, call callback with null data and the status, this includes 400/500 status codes
-          callback(null, response.status);
-        }
```
Comment (jaredvu) [inline src/lib/abacus/rest.ts:64]: Are we sure this won't affect handling of any other rest request?

Comment (rosepuppy) [reply src/lib/abacus/rest.ts:64]: I think it should be ok, the filtering is not needed. It looks like IOS also does not have logic filtering based on status code: https://github.com/dydxprotocol/v4-native-ios/blob/b5d65b38e693801f5ba0acfde1d6bb4c8ab8ba9c/dydx/dydxStateManager/dydxStateManager/Protocol%20Implementations/AbacusRestImp.swift#L67C24-L67C28

so if abacus is working well for IOS, work.

File: src/views/forms/AccountManagementForms/DepositForm.tsx [hunk 2]
```diff
@@ -300,7 +311,7 @@ export const DepositForm = ({ onDeposit, onError }: DepositFormProps) => {
     }
 
     return undefined;
-  }, [error, balance, chainId, fromAmount, sourceToken]);
+  }, [error, routeErrors, balance, chainId, fromAmount, sourceToken]);
```
Comment (jaredvu) [inline src/views/forms/AccountManagementForms/DepositForm.tsx:314]: could we add `stringGetter` as a dep

File: src/views/forms/AccountManagementForms/DepositForm.tsx [hunk 3]
```diff
@@ -287,6 +288,16 @@ export const DepositForm = ({ onDeposit, onError }: DepositFormProps) => {
       return parseWalletError({ error, stringGetter }).message;
     }
 
+    if (routeErrors) {
+      const parsedErrors = JSON.parse(routeErrors);
```
Comment (jaredvu) [inline src/views/forms/AccountManagementForms/DepositForm.tsx:292]: should we add a try/catch for this?

File: src/views/forms/AccountManagementForms/DepositForm.tsx [hunk 4]
```diff
@@ -98,7 +98,7 @@ export const DepositForm = ({ onDeposit, onError }: DepositFormProps) => {
 
   useEffect(() => {
     const hasInvalidInput =
-      debouncedAmountBN.isNaN() || debouncedAmountBN.lte(0) || debouncedAmountBN.gte(balanceBN);
```
Comment (jaredvu) [inline src/views/forms/AccountManagementForms/DepositForm.tsx:100]: good catch

**Review by jaredvu — APPROVED** [review]

File: src/views/forms/AccountManagementForms/WithdrawForm.tsx [hunk 5]
```diff
@@ -281,6 +282,21 @@ export const WithdrawForm = () => {
...
+      try {
+        const parsedErrors = JSON.parse(routeErrors);
+        if (parsedErrors?.[0]?.message) {
+          return stringGetter({
+            key: STRING_KEYS.SOMETHING_WENT_WRONG_WITH_MESSAGE,
+            params: { ERROR_MESSAGE: parsedErrors?.[0]?.message },
+          });
+        }
+      } catch (e) {
+        log('WithDrawForm/errorMessage', e);
```
Comment (aforaleka) [inline src/views/forms/AccountManagementForms/WithdrawForm.tsx:295]: nit: WithdrawForm

**Review by aforaleka — APPROVED** [review]

**Pull request merged**